design-documents/bit-precise-types.rst

```diff
@@ -194,20 +458,28 @@ It has the following negatives:
 
 Summary, suggestion, and reasoning
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Overall it seems that for operations on small values option ``A`` is more
-performant.  However, when acting on "large" values (i.e. greater than the size
-of one register) it loses some of that benefit.  Storing to and from memory
-would also come at a cost for this representation.  This is also likely to be
-the most surprising representation for developers on an Arm platform.
+
+Overall it seems that option ``A`` is more performant for operations on small
+values.  However, when acting on "large" values (i.e. greater than the size of
+one register) it loses some of that benefit.  Storing to and from memory would
+also come at a cost for this representation.  This is also likely to be the most
+surprising representation for developers on an Arm platform.
 
```

Comment:
Note there is no reason why a compiler couldn't do register operations in format A, while passing arguments/storing as C and assuming B when loading bitints or using parameters/return values. A seems most useful as an optimized internal representation (which is entirely private) rather than as the storage format.

Reply:
:+1:  I'll add a paragraph explicitly mentioning that the compiler could always choose to use a different representation at non ABI boundaries.